velox/duckdb/conversion/DuckConversion.cpp

```diff
@@ -138,7 +138,41 @@ variant duckValueToVariant(const Value& val) {
     case LogicalTypeId::DOUBLE:
       return variant(val.GetValue<double>());
     case LogicalTypeId::DECIMAL:
-      return variant(val.GetValue<double>());
+      if (handleDecimalAsDouble) {
+        return variant(val.GetValue<double>());
+      } else {
```

Comment:
this branch is rather long; consider extracting a help function

Reply:
done.